bindings/postgres/postgres.go

```diff
@@ -0,0 +1,176 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package postgres
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/pkg/errors"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+// List of operations.
+const (
+	ExecOperation  bindings.OperationKind = "exec"
+	QueryOperation bindings.OperationKind = "query"
+	CloseOperation bindings.OperationKind = "close"
+
+	ConnectionURLKey = "url"
+	CommandSQLKey    = "sql"
+
+	CommandTimeoutDefault = 5
+)
+
+// Binding represents PostgreSQL output binding
+type Binding struct {
+	logger logger.Logger
+	db     *pgxpool.Pool
+}
+
+var _ = bindings.OutputBinding(&Binding{})
```

Comment:
why do we need this ?

Reply:
that's a trick to validates adherence to interface, 0 impact at runtime

Reply:
good to learn :)